src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -166,7 +177,9 @@ def mark_lost(self, svc_id):
             id=int(svc_id),
             yes_i_really_mean_it=True)
 
-    def create(self, uuid=None, svc_id=None):
+    @Endpoint('POST')
```

Comment:
Per our previous discussion, we better removing `/orchestrator/osd` endpoint and integrate the logic into `/api/osd`.

Turn out there is already a create call in the `/api/osd` endpoint. The call uses ceph CLI to create an OSD in the cluster without any associated devices (just an OSD ID and UUID).  Besides the testing codes in QA task, I can't find any usage in the frontend. See https://github.com/bk201/ceph/blob/master/qa/tasks/mgr/dashboard/test_osd.py#L83-L98, I'm not sure if I missed something.

So I propose modifying this one to a special endpoint (`/api/osd/test_create`) since it's only used in testing. And make the `POST /api/osd` way to OSD creation with DriveGroups.

Any takes? @epuertat @p-se

Reply:
Fine with me. Personally, I'd prefer `/api/osd/create_bare` (or something similar)  as API name as it creates a bare container for an OSD, although it is only used in testing. If we were to implement it in the front-end, we wouldn't need to change the endpoint.

Reply:
`create_bare` sounds better. Thanks and fixed!

Reply:
IMHO:
* there's no need to create a new endpoint: `POST /api/osd` and `POST /api/osd, HTTP body: {bare: true}` would work.
  * we should definitely stop putting verbs in REST API endpoints and be more careful about those (`create`, `create_bare`, or `GET /cephfs/.../get_quota` and  `POST /cephfs/.../set_quota` while we already had the right `GET /rgw/user/<id>/quota` and `PUT /rgw/user/<id>/quota`).
* I'd avoid adding parameters that add little value to dashboard users. If the need of that functionality comes from testing purposes, I'd create a `TEST` flag in Dashboard (similar to the DEBUG one), and only expose this things under testing.

Reply:
Thanks. Instead of using `bare` as key, I add a `method` key to indicate the meaning of data.

*bare*
```
{
    "method": "bare",
    "data": {
        "uuid": "xxxx",
        "svc_id": 5
    }
}
```

*DriveGroups*
```
{
    "method": "drive_groups",
    "data": {
        < drive group spec here>
    }
}
```

Now we have only one endpoint for creating OSDs: `POST /api/osd`